### PR TITLE
Fix #7966: A non-anonymous type can still fail to have a symbol.

### LINKED ISSUE #7966: TypeError in `getQuickInfoAtPosition` for an intersection type
Reported in https://github.com/Microsoft/TypeScript/issues/7934#issuecomment-207359046 by @Arnavion .

```ts
class A {
    static STATIC_A: string = 'A';
    instanceA: string = 'A';
    constructor() { }
}
class B {
    static STATIC_B: string = 'B';
    instanceB: string = 'B';
}


// Define a contract to bind an instance type param (T/Z) to its static "parent" (TType/ZType)
interface ConcreteTypeOf<T> extends Function { new (...args): T; }

// Explicitly say what we are dealing with:
class MergeTypes<T, TType extends ConcreteTypeOf<T>, Z, ZType extends ConcreteTypeOf<Z>> {
    constructor(private type1: TType, private type2: ZType) { }

    // We return a constructor for T & Z and intersect static members.
    merge(): ConcreteTypeOf<T & Z> & TType & ZType {
        // Do some work to mixin the types.
        return <any>{};
    }
}

let AB = new MergeTypes<A, typeof A, B, typeof B>(A, B).merge();    // Inferred AB: ConcreteTypeOf<A & B> & typeof A & typeof B
let ab = new AB();                                                  // Inferred ab: A & B
```

hover over `AB` in `new AB()`.

```

TypeError: Unable to get property 'flags' of undefined or null reference
   at buildSymbolDisplay (:16512:17)
   at Anonymous function (:45473:13)
   at mapToDisplayParts (:45459:9)
   at symbolToDisplayParts (:45472:9)
   at getSymbolDisplayPartsDocumentationAndSymbolKind (:51614:41)
   at getQuickInfoAtPosition (:51917:13)
   at Anonymous function (:55942:119)
   at simpleForwardCall (:55793:9)
   at forwardJSONCall (:55809:13)
   at forwardJSONCall (:55855:13)
   at getQuickInfoAtPosition (:55942:13)

```

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #7966.

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -4349,7 +4349,7 @@ namespace ts {
                                         displayParts.push(keywordPart(SyntaxKind.NewKeyword));
                                         displayParts.push(spacePart());
                                     }
-                                    if (!(type.flags & TypeFlags.Anonymous)) {
+                                    if (!(type.flags & TypeFlags.Anonymous) && type.symbol) {
```
Comment (mhegazy) [inline src/services/services.ts]: why not just `if (type.symbol)`?

Comment (andy-ms) [inline src/services/services.ts]: If we allow anything with `.symbol`, anonymous function tooltips will look like `var lambdaFoo: (Anonymous function)(a: number, b: number) => number'`.

Comment (mhegazy) [inline src/services/services.ts]: i see.

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**